src/TraceEvent/TraceEvent.Tests/SpeedScopeExporterTests.cs

```diff
@@ -13,410 +19,402 @@ public class SpeedScopeStackSourceWriterTests
+            Assert.False(Validate(new[] { openEvent, closeEvent }));
+        }
 
-            Assert.Equal(ProfileEventType.Open, aggregatedEvents[0].Type);
-            Assert.Equal(0.1f, aggregatedEvents[0].RelativeTime);
-            Assert.Equal(0, aggregatedEvents[0].Depth);
+        [Fact]
+        public void ValidationDetectsIncompleteResults_InvalidOrder()
+        {
+            var openEvent = new ProfileEvent(ProfileEventType.Open, frameId: 1, 1.0f, depth: 1);
+            var closeEvent = new ProfileEvent(ProfileEventType.Close, frameId: 1, 1.0f, depth: 1);
 
-            Assert.Equal(ProfileEventType.Close, aggregatedEvents[1].Type);
-            Assert.Equal(0.1f + metric, aggregatedEvents[1].RelativeTime);
-            Assert.Equal(0, aggregatedEvents[0].Depth);
+            Assert.False(Validate(new[] { closeEvent, openEvent })); // close and then open
+        }
 
-            Assert.Equal(ProfileEventType.Open, aggregatedEvents[2].Type);
-            Assert.Equal(0.2f, aggregatedEvents[2].RelativeTime);
-            Assert.Equal(0, aggregatedEvents[2].Depth);
+        [Fact]
+        public void ValidationDetectsIncompleteResults_InvalidTimeOrder()
+        {
+            var openEvent = new ProfileEvent(ProfileEventType.Open, frameId: 1, 2.0f, depth: 1);
+            var closeEvent = new ProfileEvent(ProfileEventType.Close, frameId: 1, 1.0f, depth: 1);
 
-            Assert.Equal(ProfileEventType.Close, aggregatedEvents[3].Type);
-            Assert.Equal(0.2f + metric, aggregatedEvents[3].RelativeTime);
-            Assert.Equal(0, aggregatedEvents[3].Depth);
+            Assert.False(Validate(new[] { openEvent, closeEvent })); // close and then open
         }
 
         [Fact]
-        public void CloseMetricCanBeZeroIfItDoesNotCreateAProfileEventThatStartsAndEndsAtTheSameMoment()
+        public void ValidationAllowsForCompleteResults()
         {
-            const float metric = 0.1f;
+            var openEvent = new ProfileEvent(ProfileEventType.Open, frameId: 1, 1.0f, depth: 1);
+            var closeEvent = new ProfileEvent(ProfileEventType.Close, frameId: 1, 1.0f, depth: 1);
 
-            var samples = new[]
-            {
-                new Sample((StackSourceCallStackIndex)1, metric: metric, relativeTime: 0.1, depth: 0, callerStackIndex: 0),
-                new Sample((StackSourceCallStackIndex)1, metric: metric, relativeTime: 0.2, depth: 0, callerStackIndex: 0),
-                new Sample((StackSourceCallStackIndex)1, metric: 0.0f, relativeTime: 0.3, depth: 0, callerStackIndex: 0), // 0.0 metric
-            };
+            Assert.True(Validate(new[] { openEvent, closeEvent }));
+        }
 
-            var input = new Dictionary<int, List<Sample>>() { { 0, samples.ToList() } };
+        [Theory]
+        [InlineData("HeartRateMonitor.10068.nettrace.zip")]
+        [InlineData("VoiceMemo.23092.nettrace.zip")]
+        public void CanConverProviedTraceFiles(string zippedTraceFileName)
```

Comment:
```suggestion
        public void CanConvertProvidedTraceFiles(string zippedTraceFileName)
```
I don't mind adding my trace files but maybe we can add more sample traces here?